### ISSUE TITLE
TODO: Add title

### ISSUE THREAD

**Issue by tioteath** [koalaman/shellcheck#1894]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2016
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh

fswatch . --recursive --utc-time --one-per-batch --insensitive --allow-overflow --print0 --monitor=poll_monitor | xargs -0 -I{} --no-run-if-empty /bin/sh -c 'export ACTUAL_PWD=${PWD}'
```

#### Here's what shellcheck currently says:
```
Line 3	SC2016: Expressions don't expand in single quotes, use double quotes for that.
```
#### Here's what I wanted or expected to see:

Expressions get expanded in the fork of `sh`, hence single quotes.

**Comment by Akianonymus** [comment]
+1

**Comment by dimo414** [comment]
@tioteath can you fix the issue title?

**Comment by pmenzel** [comment]
Same issue with `bash -c 'some bash commands'`

**Comment by dimo414** [comment]
IMO this finding is correct, and should be [suppressed](https://github.com/koalaman/shellcheck/wiki/SC2016#exceptions) if no variable expansion is intended. Callers could just as easily _want_ eager variable expansion, and single-quotes + a suppression makes it clear the author intended to lazily evaluate the variables in the subprocess' environment.

There is an [allowlist](https://github.com/koalaman/shellcheck/blob/08d2eef4111afde4e24a0c7860ce9d8725db37a6/src/ShellCheck/Analytics.hs#L805) for commands that often expect single-quoted variables, but I worry `xargs` would be too heavy-handed a command to include in that list.

**Comment by nwithers-ecr** [comment]
Here is something similar that I am doing
```bash
xargs -I{} bash -c 'k3d kubeconfig get {} > $HOME/.kube/config.d/k3d-{}.yaml'
```
but rather than not displaying SC2016 (we can always turn it off), it seems to me these expressions should be reporting  SC2086
